gax-grpc/src/test/java/com/google/api/gax/grpc/GrpcClientCallsTest.java

```diff
@@ -75,4 +82,48 @@ public void testAffinity() {
     assertThat(gotCallA).isSameAs(gotCallB);
     assertThat(gotCallA).isNotSameAs(gotCallC);
   }
+
+  @Test
+  public void testExtraHeaders() {
+    Metadata emptyHeaders = new Metadata();
+    final Map<String, List<String>> extraHeaders = new HashMap<>();
+    extraHeaders.put(
+        "header-key-1", ImmutableList.<String>of("header-value-11", "header-value-12"));
+    extraHeaders.put("header-key-2", ImmutableList.<String>of("header-value-21"));
+
+    MethodDescriptor<Color, Money> descriptor = FakeServiceGrpc.METHOD_RECOGNIZE;
+
+    @SuppressWarnings("unchecked")
+    ClientCall<Color, Money> mockClientCall = Mockito.mock(ClientCall.class);
+
+    @SuppressWarnings("unchecked")
+    ClientCall.Listener<Money> mockListener = Mockito.mock(ClientCall.Listener.class);
+
+    @SuppressWarnings("unchecked")
+    Channel mockChannel = Mockito.mock(ManagedChannel.class);
+
+    Mockito.doAnswer(
+            new Answer<Void>() {
```

Comment:
Formatting feels wrong (it is 8 spaces indentation).

Reply:
gradlew goJF seems to like this format, probably we need to adjust it there?